packages/runtime/src/widgets/Form/TextInput.tsx

```diff
@@ -58,6 +58,30 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
     [action],
   );
 
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (values?.inputType !== "number") {
+        return;
+      }
+
+      const isNumericKey = /^\d$/.test(e.key); // Numpad keys (96-105)
+      const isControlKey = [
+        "Backspace",
+        "Tab",
+        "ArrowLeft",
+        "ArrowRight",
+      ].includes(e.key);
+      const isCopyPasteCut =
+        (e.ctrlKey || e.metaKey) &&
+        ["c", "v", "x"].includes(e.key.toLowerCase());
```

Comment:
I think just check if the meta/ctrl key is pressed and let the default behavior go through if so.